main.go

```diff
@@ -110,14 +155,153 @@ func runMain() error {
+	}
+
+	go wait()
+	fmt.Printf("VERIFICATION: Release new version '%s'? (ENTER to continue, Ctrl-C to quit))", nextReleaseVersion.String())
+	fmt.Scanln()
+
+	undoChanges := true
+	defer func() {
+		if undoChanges {
+			// git reset --hard origin/master
+			originMasterCommitHash, err := repository.ResolveRevision(plumbing.Revision(remoteMasterBranchName))
+			err = worktree.Reset(&git.ResetOptions{Mode: git.HardReset, Commit: *originMasterCommitHash })
+			if err != nil {
+				logrus.Errorf("ACTION REQUIRED: Error occurred attempting to undo local changes made for release '%s'. Please run 'git reset --hard origin/master' to undo manually.", nextReleaseVersion.String(), err)
+			}
+		}
+	}()
+
+	fmt.Println("Running prerelease scripts...")
+	preReleaseScriptsDirpath := path.Join(currentWorkingDirectory, relScriptsDirpath)
+	err = runPreReleaseScripts(preReleaseScriptsDirpath, nextReleaseVersion.String())
+	if err != nil {
+		return stacktrace.Propagate(err, "An error occurred while running prerelease scripts.")
+	}
+
+	fmt.Println("Updating the changelog...")
+	err = updateChangelog(changelogFilepath, nextReleaseVersion.String())
+	if err != nil {
+		return stacktrace.Propagate(err, "An error occurred while updating the changelog.")
+	}
+
+	fmt.Println("Committing changes locally...")
+	// Commit pre release changes
+	err = worktree.AddWithOptions(&git.AddOptions{All: true})
+	if err != nil {
+		return stacktrace.Propagate(err, "An error occurred while attempting to add release changes to staging area.")
+	}
+	commitMsg := fmt.Sprintf("Finalize changes for release version '%s'", nextReleaseVersion.String())
+	_, err = worktree.Commit(commitMsg, &git.CommitOptions{
+		Author: &object.Signature{
+			Name:  name,
+			Email: email,
+			When:  time.Now(),
+		},
+	})
+
+	releaseTag := nextReleaseVersion.String()
+	vReleaseTag := fmt.Sprintf("v.%s", nextReleaseVersion.String())
+
+	undoReleaseTag := true
+	defer func() {
+		if undoReleaseTag {
+			// git tag -d
+			err = repository.DeleteTag(releaseTag)
+			if err!= nil {
+				logrus.Errorf("ACTION REQUIRED: Error occurred attempting to undo tag '%s'. Please run 'git tag -d %s' to delete the tag manually.", releaseTag, err)
+			}
+			err = repository.DeleteTag(vReleaseTag)
+			if err != nil {
+				logrus.Errorf("ACTION REQUIRED: Error occurred attempting to undo tag '%s'. Please run 'git tag -d %s' to delete the tag manually.", vReleaseTag, err)
+			}
+		}
+	}()
+
+	fmt.Println("Setting next release version tag...")
+	// Set next release version tag
+	head, err := repository.Head()
+	if err != nil {
+		return stacktrace.Propagate(err, "An error occurred while attempting to get the ref to HEAD of the local repository.")
+	}
+	_, err = repository.CreateTag(releaseTag, head.Hash(), &git.CreateTagOptions{
+		Message: releaseTag,
+	})
+	if err != nil {
+		return stacktrace.Propagate(err, "An error occurred while attempting to create a git tag for the next release version.")
```

Comment:
Careful, you don't have the `releaseTag` in this log message and you have an identical log message below so a dev could get confused about which fired. Putting the tag name in this and the below will fix